src/data_availability/node_state.rs

```diff
@@ -66,43 +79,56 @@ impl NodeState {
                     }
                 }
                 TransactionData::Proof(_) => {
-                    error!("Unverified proof transaction should not be in a block");
-                }
-                TransactionData::RecursiveProof(_) => {
                     error!("Unverified recursive proof transaction should not be in a block");
                 }
-                TransactionData::VerifiedProof(verified_proof_transaction) => {
-                    match self.handle_verified_proof_tx(verified_proof_transaction) {
-                        Ok(proof_tx_output) => {
-                            // When a proof tx is handled, three things happen:
-                            // 1. Blobs get verified
-                            // 2. Maybe: BlobTransactions get settled
-                            // 3. Maybe: Contract state digests are updated
-
-                            // Keep track of verified blobs
-                            verified_blobs.push((
-                                verified_proof_transaction.hyle_output.tx_hash.clone(),
-                                verified_proof_transaction.hyle_output.index.clone(),
-                            ));
-                            // Keep track of settled txs
-                            settled_blob_tx_hashes.extend(proof_tx_output.settled_blob_tx_hashes);
-                            // Update the contract's updated states
-                            updated_states.extend(proof_tx_output.updated_states);
-                            // Keep track of all verified proof txs
-                            new_verified_proof_txs.push(tx.clone());
-                        }
-                        Err(e) => {
-                            error!("Failed to handle proof transaction: {:?}", e);
-                            failed_txs.push(tx.clone());
-                        }
-                    }
-                }
-                TransactionData::VerifiedRecursiveProof(rec_proof_tx) => {
+                TransactionData::VerifiedProof(rec_proof_tx) => {
+                    let mut rec_metadata_iter = if let Some(meta) = &rec_proof_tx.recursive_metadata
+                    {
+                        meta.iter()
+                    } else {
+                        std::slice::Iter::default()
+                    };
```

Comment:
Je trouve ça un peu alambiqué de créer un iterateur ici
On peut pas itérer direct dessus?

Reply:
Nope parce que c'est une option, et il me faut une valeur pour le "else" sinon ça prend la tête (je le unwrap plus bas).

Je peux ptet faire un peu mieux, je vais revoir, mais c'est un poil relou.